src/components/engine/engine-instances-table.tsx

```diff
@@ -397,36 +410,124 @@ const RemoveModal = ({
+      });
+    }
+  };
+
   return (
     <Modal isOpen onClose={onClose} isCentered size="lg">
       <ModalOverlay />
       <ModalContent>
-        <ModalHeader>Remove Engine Instance</ModalHeader>
-
         {instance.cloudDeployedAt ? (
           <>
-            <ModalBody>
+            <ModalHeader>Cancel Engine Subscription</ModalHeader>
+
+            <ModalBody as={Stack} gap={4}>
               <Text>
-                To cancel your Engine subscription, please contact{" "}
-                <Link
-                  href="mailto:support@thirdweb.com"
-                  isExternal
-                  color="primary.500"
-                >
-                  support@thirdweb.com
-                </Link>
-                .
+                Complete this form to request to cancel your Engine
+                subscription. This may take up to 2 business days.
               </Text>
+
+              {/* Form */}
+              <FormControl>
+                <FormLabel>
+                  Please share any feedback to help us improve
+                </FormLabel>
+                <RadioGroup>
+                  <Stack>
+                    <Radio
+                      value="USING_SELF_HOSTED"
+                      {...form.register("reason", { required: true })}
+                    >
+                      <Text>Migrating to self-hosted</Text>
+                    </Radio>
+                    <Radio
+                      value="TOO_EXPENSIVE"
+                      {...form.register("reason", { required: true })}
+                    >
+                      <Text>Too expensive</Text>
+                    </Radio>
+                    <Radio
+                      value="MISSING_FEATURES"
+                      {...form.register("reason", { required: true })}
+                    >
+                      <Text>Missing features</Text>
+                    </Radio>
+                    <Radio
+                      value="OTHER"
+                      {...form.register("reason", { required: true })}
+                    >
+                      <Text>Other</Text>
+                    </Radio>
+                  </Stack>
+                </RadioGroup>
+                <Textarea
+                  placeholder="Provide additional feedback"
+                  mt={2}
+                  {...form.register("feedback")}
+                />
+              </FormControl>
+
+              <Alert status="warning" variant="left-accent">
+                <Flex direction="column" gap={2}>
+                  <Heading as={AlertTitle} size="label.md">
+                    This action is irreversible!
+                  </Heading>
+                  <Text as={AlertDescription} size="body.md">
+                    You will no longer be able to access this Engine&apos;s
+                    local backend wallets.{" "}
+                    <strong>Any remaining mainnet funds will be lost.</strong>
```

Comment:
all network (testnet & mainnet) funds will be lost but I guess `mainnet` makes sense here.